src/apply.jl

```diff
@@ -61,58 +61,43 @@ function apply!(A::AbstractArray, t::Transform; kwargs...)
 end
 
 """
-    apply(table, ::Transform; cols=nothing, [header], kwargs...) -> Table
+    apply(table, ::Transform; [cols], [header], kwargs...) -> Table
 
 Applies the [`Transform`](@ref) to each of the specified columns in the `table`.
 If no `cols` are specified, then the [`Transform`](@ref) is applied to all columns.
 
 Optionally provide a `header` for the output table. If none is provided the default in
 `Tables.table` is used.
 """
-function apply(table, t::Transform; cols=nothing, kwargs...)
+function apply(table, t::Transform; cols=_get_cols(table), kwargs...)
     Tables.istable(table) || throw(MethodError(apply, (table, t)))
 
     # Extract a columns iterator that we should be able to use to mutate the data.
     # NOTE: Mutation is not guaranteed for all table types, but it avoid copying the data
-    columntable = Tables.columns(table)
-    cnames = cols === nothing ? propertynames(columntable) : cols
+    coltable = Tables.columntable(table)
+    cols = _to_vec(cols)
 
-    result = _apply(columntable, t, cnames; kwargs...)
+    result = reduce(hcat, [_apply(getproperty(coltable, col), t; kwargs...) for col in cols])
     header = get(kwargs, :header, nothing)
```

Comment:
Is there a reason we extract `header` this way as opposed to setting `header=nothing` in the function signature?

Reply:
I was trying to be too smart about it and wrap it in the non-`apply` kwargs. Now that you say that it's just as bad because it's wrapping it in the `Transform` kwargs. Better to be explicit about it.